src/core/github-api.js

```diff
@@ -0,0 +1,126 @@
+    Accept: "application/vnd.github.v3.html+json",
+  };
+  if (githubUser && githubToken) {
+    const credentials = btoa(`${githubUser}:${githubToken}`);
+    const Authorization = `Basic ${credentials}`;
+    Object.assign(headers, { Authorization });
+  } else if (githubToken) {
+    const Authorization = `token ${githubToken}`;
+    Object.assign(headers, { Authorization });
+  }
+  return headers;
+}
+
+export async function getRateLimit(conf) {
+  const headers = githubRequestHeaders(conf);
+  const request = new Request(`https://api.github.com/rate_limit`, {
+    mode: "cors",
+    referrerPolicy: "no-referrer",
+    headers,
+  });
+  const response = await fetch(request);
+  const responseJSON = await response.json();
+  return responseJSON.rate.remaining;
+}
+
+export async function fetchAndStoreGithubIssues(conf) {
+  const { githubAPI } = conf;
+  /** @type {NodeListOf<HTMLElement>} */
+  const specIssues = document.querySelectorAll(".issue[data-number]");
+  let remainingRequests = await getRateLimit(conf);
+  if (specIssues.length > remainingRequests) {
+    const msg =
+      `Your spec contains ${specIssues.length} Github issues, ` +
+      `but your current GitHub quota only allows ${remainingRequests} more requests. Some issues will not show up`;
+    pub("warning", msg);
+  }
+  const issuePromises = [...specIssues]
+    .map(elem => Number.parseInt(elem.dataset.number, 10))
+    .filter(issueNumber => issueNumber)
+    .map(async issueNumber => {
+      if (remainingRequests > 0) {
+        const issueURL = `${githubAPI}/issues/${issueNumber}`;
+        const headers = githubRequestHeaders(conf);
+        const request = new Request(issueURL, {
+          mode: "cors",
+          referrerPolicy: "no-referrer",
+          headers,
+        });
+        const response = await fetchAndCache(request);
+        remainingRequests--;
+        return processResponse(response, issueNumber);
+      }
+    });
+  const issues = await Promise.all(issuePromises);
+  return new Map(issues);
+}
+
+/**
+ * @typedef {{ color: string, name: string }} GitHubLabel
+ * @typedef {{ title: string, number: number, state: string, message: string, body_html: string, labels: GitHubLabel[] }} GitHubIssue
+ *
+ * @param {Response} response
+ * @param {number} issueNumber
+ */
+async function processResponse(response, issueNumber) {
+  // "message" is always error message from GitHub
+  const issue = { title: "", number: issueNumber, state: "", message: "" };
+  try {
+    const json = await response.json();
+    Object.assign(issue, json);
+  } catch (err) {
+    issue.message = `Error JSON parsing issue #${issueNumber} from GitHub.`;
+  }
+  if (!response.ok || issue.message) {
+    const msg = `Error fetching issue #${issueNumber} from GitHub. ${
+      issue.message
+    } (HTTP Status ${response.status}).`;
+    pub("error", msg);
+  }
+  return /** @type {[number, GitHubIssue]} */ ([issueNumber, issue]);
```

Comment:
Nit: We should move the in JSDoc comment above as `@returns`, instead of casting here.